File: core/src/main/java/com/xmartlabs/bigbang/core/model/BuildInfo.java
package com.xmartlabs.bigbang.core.model;

/** Defines a set of useful information about the current build. */
public interface BuildInfo {
  /**
   * Checks whether or not the current build is in debug mode.
   *
   * @return true if the build is in debug
   */
  boolean isDebug();

  /**
   * Checks whether or not the current environment is staging.
   *
   * @return true if the environment is staging
   */
  boolean isStaging();

  /**
   * Checks whether or not the current environment is production.
   *
   * @return true if the environment is production
   */
  boolean isProduction();
}

File: core/src/main/java/com/xmartlabs/bigbang/core/model/BuildInfo.kt
package com.xmartlabs.bigbang.core.model

/** Defines a set of useful information about the current build.  */
interface BuildInfo {
  /** Checks whether or not the current build is in debug mode. */
  val isDebug: Boolean

  /** Checks whether or not the current environment is staging. */
  val isStaging: Boolean

  /** Checks whether or not the current environment is production. */
  val isProduction: Boolean
}
